phd-tests/framework/src/guest_os/windows.rs

```diff
@@ -14,52 +14,60 @@ use super::{CommandSequence, CommandSequenceEntry, GuestOsKind};
 /// - Cygwin is installed to C:\cygwin and can be launched by invoking
 ///   C:\cygwin\cygwin.bat.
 /// - The local administrator account is enabled with password `0xide#1Fan`.
-pub(super) fn get_login_sequence_for(guest: GuestOsKind) -> CommandSequence {
+pub(super) fn get_login_sequence_for<'a>(
+    guest: GuestOsKind,
+) -> CommandSequence<'a> {
     assert!(matches!(
         guest,
-        GuestOsKind::WindowsServer2019 | GuestOsKind::WindowsServer2022
+        GuestOsKind::WindowsServer2016
+            | GuestOsKind::WindowsServer2019
+            | GuestOsKind::WindowsServer2022
     ));
 
     let mut commands = vec![
         CommandSequenceEntry::WaitFor(
-            "Computer is booting, SAC started and initialized.",
+            "Computer is booting, SAC started and initialized.".into(),
         ),
         CommandSequenceEntry::WaitFor(
-            "EVENT: The CMD command is now available.",
+            "EVENT: The CMD command is now available.".into(),
         ),
-        CommandSequenceEntry::WaitFor("SAC>"),
-        CommandSequenceEntry::WriteStr("cmd"),
-        CommandSequenceEntry::WaitFor("Channel: Cmd0001"),
-        CommandSequenceEntry::WaitFor("SAC>"),
-        CommandSequenceEntry::WriteStr("ch -sn Cmd0001"),
+        CommandSequenceEntry::WaitFor("SAC>".into()),
+        CommandSequenceEntry::WriteStr("cmd".into()),
+        CommandSequenceEntry::WaitFor("Channel: Cmd0001".into()),
+        CommandSequenceEntry::WaitFor("SAC>".into()),
+        CommandSequenceEntry::WriteStr("ch -sn Cmd0001".into()),
         CommandSequenceEntry::WaitFor(
-            "Use any other key to view this channel.",
+            "Use any other key to view this channel.".into(),
         ),
-        CommandSequenceEntry::WriteStr(""),
-        CommandSequenceEntry::WaitFor("Username:"),
-        CommandSequenceEntry::WriteStr("Administrator"),
-        CommandSequenceEntry::WaitFor("Domain  :"),
-        CommandSequenceEntry::WriteStr(""),
-        CommandSequenceEntry::WaitFor("Password:"),
-        CommandSequenceEntry::WriteStr("0xide#1Fan"),
+        CommandSequenceEntry::WriteStr("".into()),
+        CommandSequenceEntry::WaitFor("Username:".into()),
+        CommandSequenceEntry::WriteStr("Administrator".into()),
+        CommandSequenceEntry::WaitFor("Domain  :".into()),
+        CommandSequenceEntry::WriteStr("".into()),
+        CommandSequenceEntry::WaitFor("Password:".into()),
+        CommandSequenceEntry::WriteStr("0xide#1Fan".into()),
```

Comment:
nit, take it or leave it: maybe worth adding functions like
```rust
impl<'a> CommandSequenceEntry<'a> {
    pub fn write_str(s: impl Into<Cow<'a, str>>) -> Self {
         Self::WriteStr(s.into())
    }

    pub fn wait_for(s: impl Into<Cow<'a, str>>) -> Self {
         Self::WaitFor(s.into())
    }
}
```
and then calling them instead of having to `.into()` everything? 

not a big deal, but maybe a little bit nicer?

Reply:
I like this much better, thanks! The `.into()` grossness was part of made me question whether `Cow` was the right tool for this job.